gazebo_ros/src/node.cpp

```diff
@@ -164,4 +183,24 @@ rclcpp::Logger Node::internal_logger()
   return rclcpp::get_logger("gazebo_ros_node");
 }
 
+void NodeLookUp::add_node(const std::string& node_name, Node::SharedPtr ros_node) {
+  std::lock_guard<std::mutex> guard(this->internal_mutex_);
+  this->map_.emplace(node_name, ros_node);
+}
+
+Node::SharedPtr NodeLookUp::get_node(const std::string& node_name) {
```

Comment:
Now that I think about it, we don't require a ``get_node`` method in this container, as we are not going to return a node with the name name if that exists already. In fact, we can change the container to just a ``std::vector`` instead of a ``map``

Reply:
Thanks, makes sense. 
Instead of `vector` maybe we can have `unordered_set` I guess? Probably will be more efficient for search but we can discuss that later as well.

Reply:
Sound good to me, that would be more efficient.